cmd/dump.go

```diff
@@ -76,3 +95,64 @@ func dumpInstalledObjectsCmd(cmd *cobra.Command, args []string) error {
 	cmd.Printf("Dumped %d installed objects for package %s to %s\n", n, packageName, outputPath)
 	return nil
 }
+
+func dumpAgentPoliciesCmdAction(cmd *cobra.Command, args []string) error {
+	packageName, err := cmd.Flags().GetString(cobraext.PackageFlagName)
+	if err != nil {
+		return cobraext.FlagParsingError(err, cobraext.PackageFlagName)
+	}
+
+	agentPolicy, err := cmd.Flags().GetString(cobraext.AgentPolicyFlagName)
+	if err != nil {
+		return cobraext.FlagParsingError(err, cobraext.AgentPolicyFlagName)
+	}
+
+	outputPath, err := cmd.Flags().GetString(cobraext.DumpOutputFlagName)
+	if err != nil {
+		return cobraext.FlagParsingError(err, cobraext.DumpOutputFlagName)
+	}
+
+	tlsSkipVerify, _ := cmd.Flags().GetBool(cobraext.TLSSkipVerifyFlagName)
+
+	var clientOptions []kibana.ClientOption
+	if tlsSkipVerify {
+		clientOptions = append(clientOptions, kibana.TLSSkipVerify())
+	}
+	kibanaClient, err := kibana.NewClient(clientOptions...)
+	if err != nil {
+		return errors.Wrap(err, "failed to initialize Kibana client")
+	}
+
+	switch {
+	case agentPolicy != "":
+		dumper := dump.NewAgentPoliciesDumper(kibanaClient, &agentPolicy)
+		err = dumper.DumpAgentPolicy(cmd.Context(), outputPath)
```

Comment:
Nit. This API is using different approaches to dump per agent policy or per package name. For agent policy the parameter is passed on the constructor, and for package name in the dump method.

I would suggest to use a consistent approach for both ways of duping policies.

For example it could be like this:
```
dumper := dump.NewAgentPoliciesDumper(kibanaClient)
err = dumper.DumpAgentPolicy(cmd.Context(), outputPath, dump.AgentPolicyOptions{Name: agentPolicy})
...
err = dumper.DumpAgentPolicy(cmd.Context(), outputPath, dump.AgentPolicyOptions{PackageName: packageName})
```
Or with explicit methods:
```
dumper := dump.NewAgentPoliciesDumper(kibanaClient)
err = dumper.DumpByName(cmd.Context(), outputPath, agentPolicy)
...
err = dumper.DumpByPackage(cmd.Context(), outputPath, packageName)
...
err = dumper.DumpAll(cmd.Context(), outputPath)
```
Or initialize the dumper for the given configuration:
```
dumper := dump.NewAgentPoliciesDumper(kibanaClient, agentPolicy, packageName)
err = dumper.Dump(cmd.Context())
```

Reply:
You're right, there was a mix of different approaches in these calls. Updated using explicit methods.